featuretools/primitives/standard/datetime_transform_primitives.py

```diff
@@ -331,6 +334,69 @@ def is_leap_year(vals):
         return is_leap_year
 
 
+class IsLunchTime(TransformPrimitive):
+    """Determines if a datetime falls during lunch time (non-holiday weekday with hour=12)
+
+    Args:
+        country (str): Country to use for determining Holidays.
+            Default is 'US'. Should be one of the available countries here:
+            https://github.com/dr-prodigy/python-holidays#available-countries
+        include_weekend (bool): Include weekend dates as datetimes with valid
+            lunch times. Default is True.
+        include_holidays (bool): Include holiday dates as datetimes with valid
+            lunch times. Default is False.
+
+    Examples:
+        >>> from datetime import datetime
+        >>> dates = [datetime(2022, 6, 21, 12, 3, 3),
+        ...          datetime(2019, 1, 3, 4, 4, 4),
+        ...          datetime(2022, 1, 1, 12, 1, 2),
+        ...          np.nan]
+        >>> ilt = IsLunchTime()
+        >>> ilt(dates).tolist()
+        [True, False, False, False]
+    """
+
+    name = "is_lunch_time"
+    input_types = [ColumnSchema(logical_type=Datetime)]
+    return_type = ColumnSchema(logical_type=BooleanNullable)
+    compatibility = [Library.PANDAS, Library.DASK, Library.SPARK]
+    description_template = "whether {} falls during lunch time"
+
+    def __init__(self, country="US", include_weekends=True, include_holidays=False):
+        self.country = country
+        years_list = [1950 + x for x in range(150)]
+        self.federal_holidays = getattr(holidays, self.country)(years=years_list)
+        self.include_weekends = include_weekends
+        self.include_holidays = include_holidays
+        self.holidays_df = pd.DataFrame(
+            sorted(self.federal_holidays.items()), columns=["dates", "names"]
+        )
+
+    def spark_mask(self, v):
+        l_time = v.hour == 12
+        if not self.include_weekends:
+            l_time = l_time and (v.dayofweek < 5)
+        if not self.include_holidays:
+            l_time = l_time and not (v.normalize().isin(self.holidays_df.dates))
+        return l_time
+
+    def get_function(self):
+        def is_lunch_time(vals):
+            if isinstance(vals, pd.Series) or isinstance(vals, dd.dataframe.Series):
+                mask = vals.dt.hour == 12
```

Comment:
I like this, but use better names.  `is_12_mask` and `is_weekday_mask`: then you have `mask = is_12_mask & is_weekday_mask`

Reply:
Got it. I separated out the mask creation now, with more descriptive names